Assets/Character/AaronTestAssets/PlayerMovement.cs

```diff
@@ -6,79 +6,108 @@
 public class PlayerMovement : MonoBehaviour {
 
 
-    public float speed = 2f;
-    public float jump = 8f;
+    public float move_speed = 2f;
+    public float jump_power = 8f;
+    public float roll_speed = 4f;
+    public float roll_time = 0.5f;
+
+
     private Animator animator;
     private Rigidbody2D rigidbody;
-    private bool isFacingRight = true;
-	// Use this for initialization
+    private int directionFacing = 1;
+    //private ArrayList lockStates = new ArrayList();
+    private AnimatorStateInfo stateInfo;
+    private Vector3 scale;
+    private float timer;
+	
+
 	void Start () {
 
         animator = GetComponent<Animator>();
         rigidbody = GetComponent<Rigidbody2D>();
-
-        // if (animator.)
+       // lockStates.Add();
+        stateInfo = animator.GetCurrentAnimatorStateInfo(0);
+        scale = transform.localScale;
 
     }
 	
-	// Update is called once per frame
 	void FixedUpdate () {
-
-        float horizontal = (Input.GetAxis("Horizontal"));
-
-        //position based
-        transform.position += new Vector3(horizontal, 0, 0) * speed * Time.deltaTime;
-
-        //add force
-        //rigidbody.AddForce(new Vector2(horizontal * speed, 0), ForceMode2D.Force);
-
-        if(animator.GetBool("OnGround"))
+        if (!animator.GetBool("Rolling"))
         {
-            if (horizontal != 0)
-            {
 
-                animator.Play("Walk");
-                Flip(horizontal);
+            float horizontal = (Input.GetAxis("Horizontal"));
+
+            //position based
+            transform.position += new Vector3(horizontal, 0, 0) * move_speed * Time.deltaTime;
+
+            //add force
+            //rigidbody.AddForce(new Vector2(horizontal * speed, 0), ForceMode2D.Force);
+
+                if (animator.GetBool("OnGround"))
+                {
+                    if (horizontal != 0)
+                    {
+                        animator.Play("Walk");
+                        Flip(horizontal);
+                    }
+
+                    else
+                    {
+                        animator.Play("Idle");
+                    }
+
+                    if (Input.GetKeyDown("space"))
+                    {
+                        animator.SetBool("OnGround", false);
+                        animator.Play("Jump");
+                        rigidbody.AddForce(new Vector2(0, jump_power), ForceMode2D.Impulse);
```

Comment:
use the predefined vectors 
https://docs.unity3d.com/ScriptReference/Vector2.html
see "static variables"